src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Text,
+  Image,
+  Container,
+  Divider,
+  Title,
+  Badge,
+  Group,
+  Button,
+  Space,
+} from "@mantine/core";
+import { useState, useEffect } from "react";
+import { getSingleArticle, updateVote } from "../../api.js";
+import { useParams } from "react-router-dom";
+import Loading from "../Loading";
+import Error from "../Error";
+import Comments from "./Comments.jsx";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [error, setError] = useState(undefined);
+  const [article, setArticle] = useState(undefined);
+
+  const handleUpVote = () => {
+    updateVote(article_id, 1)
+      .then((articleResponse) => {
+        setArticle(articleResponse);
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  };
+
+  const handleDownVote = () => {
+    updateVote(article_id, -1)
+      .then((articleResponse) => {
+        setArticle(articleResponse);
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  };
```

Comment:
We could have a single handleVote() function here which takes num (either 1 or -1) as an argument